server/routes/newspaperArchive.ts

```diff
@@ -0,0 +1,69 @@
+import { Router } from 'express';
+import type { Request, Response } from 'express';
+import fetch from 'node-fetch';
+import { s3ConfigPromise } from '../awsIntegration';
+import { log } from '../log';
+import { withIdentity } from '../middleware/identityMiddleware';
+
+type NewspapersRequestBody = {
+	expires?: number; // defaults to 24 hours
+	'query-string'?: string;
+};
+
+// { url: "https://<subdomain>.newspapers.com/…?tpa=<token>" }
+type NewspapersResponseBody = {
+	url: string;
+};
+
+type NewspaperArchiveConfig = {
+	authKey: string;
+};
+
+function base64(input: string) {
+	return Buffer.from(input).toString('base64');
+}
+
+export const newspaperArchiveConfigPromise: Promise<
+	NewspaperArchiveConfig | undefined
+> = s3ConfigPromise<NewspaperArchiveConfig>('authKey')('newspaper-archive');
+
+let authKey: string;
+
+const router = Router();
+
+router.use(withIdentity(401));
+
+router.get('/auth', async (_req: Request, res: Response) => {
+	const subdomain = 'theguardian';
+
+	if (authKey === undefined) {
+		const config = await newspaperArchiveConfigPromise;
```

Comment:
this isn't in a synchronised block so I assume it can cope with the expected number of requests all hitting this promise at the same time?  Or is the `newspaperArchiveConfigPromise` a singleton anyway? in which case we could just get rid of the guard completely?

Reply:
Ah, you're right it's unnecessary. The S3 config gets requested when the server starts up. I'm not entirely sure how?